webui/src/lib/components/repository/ChangeSummary.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {useEffect, useState} from "react";
+import {
+    ClockIcon,
+    DiffAddedIcon,
+    DiffIgnoredIcon,
+    DiffModifiedIcon,
+    DiffRemovedIcon,
+    XCircleIcon
+} from "@primer/octicons-react";
+import {OverlayTrigger, Tooltip} from "react-bootstrap";
+import {humanSize} from "./tree";
+
+const MAX_NUM_OBJECTS = 10_000;
+const PAGE_SIZE = 1_000;
+
+class SummaryEntry {
+    constructor() {
+        this.count = 0
+        this.size = 0
+    }
+    add(count, size) {
+        this.count += count
+        this.size += size
+    }
+}
+
+class SummaryData {
+    constructor() {
+        this.added = new SummaryEntry()
+        this.changed = new SummaryEntry()
+        this.removed = new SummaryEntry()
+        this.conflict = new SummaryEntry()
+    }
+}
+
+/**
+ * Widget to display a summary of a change: the number of added/changed/deleted/conflicting objects.
+ * Shows an error if the change is has more than {@link MAX_NUM_OBJECTS} entries.
+
+ * @param {string} prefix - prefix to display summary for.
+ * @param {(after : string, path : string, useDelimiter :? boolean, amount :? number) => Promise<any> } getMore - function to use to get the change entries.
+ */
+export default ({prefix, getMore}) => {
+    const [resultsState, setResultsState] = useState({results: [], pagination: {}, tooManyPages: false});
+    const [loading, setLoading] = useState(true);
+    useEffect(async () => {
+        // get pages until reaching the max change size
+        if (resultsState.results && resultsState.results.length >= MAX_NUM_OBJECTS) {
+            setResultsState({results: null, pagination: {}, tooManyPages: true})
+            setLoading(false)
+            return
+        }
+        if (!loading) {
+            return
+        }
+        const {results, pagination} = await getMore(resultsState.pagination.next_offset || "", prefix, false, PAGE_SIZE)
+        if (!pagination.has_more) {
+            setLoading(false)
+        }
+        setResultsState({results: resultsState.results.concat(results), pagination: pagination, tooManyPages: false})
+    }, [resultsState.results, loading])
+
+    if (loading) return <ClockIcon/>
```

Comment:
Why take this approach instead of showing results so far and keep update it forever..?

Reply:
I think it's not a good experience to show an updating number of objects.